debugging.rst

```diff
@@ -0,0 +1,101 @@
+.. _debugging:
+
+==============
+Debugging Jobs 
+==============
+
+Flux supports parallel debuggers such as Rogue Wave Software (RWS)'s
+TotalView parallel debugger (https://totalview.io).
+
+----------------------------------
+Parallel Debugging using TotalView
+----------------------------------
+
+Debugging your MPI job from the beginning of
+parallel program execution:
+
+.. code-block:: console
+
+  $ totalview --args flux mini run -N 2 -n 2 ./mpi-program
+
+Attaching to an already running job:
+
+.. code-block:: console
+
+  $ flux job attach --debug <JOBID> &
+  $ PID=$!
+  $ totalview -pid ${PID} /proc/${PID}/exe
+
+You can also just type ``totalview`` without ``-pid`` option and
+then use its ``Start a Debugging Session``
+GUI option to select the process of this ``flux-job`` command:
+``A running program (attach)``.
+
+.. note::
+  You use TotalView with the newly invoked ``flux job attach``
+  when your job has been launched via ``flux mini run`` (or ``flux mini submit``).
+  This is because Flux currently does not allow TotalView
+  to attach to your running code if you just attach it
+  to the ``flux mini run`` process that controls your code. For
+  performance reasons, ``flux mini run`` does not generate
+  sufficient debug information when it is invoked with no tool control.
+
+Flux provides ``MPIR_partial_attach_ok`` `[1] <https://www.mpi-forum.org/docs/mpir-specification-10-11-2010.pdf>`_
+so you can attach TotalView to a subset of parallel processes as well, which
+can be handy when you debug a large-scale job. Please refer to
+`TotalView user guide`_ on its subset-attach capability.
+
+.. _TotalView user guide: https://docs.roguewave.com/en/totalview/current/html/
+
+----------------------------------------------
+Better Handling of Flux's Internal Exec Events
+----------------------------------------------
+
+Flux executes a series of subcommands before launching
+your parallel program.
+Thus, a side effect of the above commands is that TotalView
+will stop Flux's own program execution each time
+a new subcommand is ``exec(3)``'ed (e.g., when ``flux`` ``exec``'s
+``flux-mini``) before your parallel program processes are spawned.
+This means you must manually click "Yes" whenever a stop-at-exec
+dialogue window is popped up. You do not need to stop
+it unless you are a Flux developer!
+Instead, you can make TotalView automatically
+continue on these events by building the the following
+exec-handling Tcl code into TotalView.
+
+.. code-block::
+
+    catch {dset TV::exec_handling { {^(flux|lrun|srun|jsrun)(<python[^>]*>(<flux-job>)?|<bash>|<jsrun>|<jswrap>)*$ go}} }
```

Comment:
Unfortunately, this block is not rendering in readthedocs right now.  I wonder if it is because the language is missing (although the docs say in Sphinx 2.0 it became optional):

```suggestion
exec-handling Tcl code into TotalView.

.. code-block:: tcl

    catch {dset TV::exec_handling { {^(flux|lrun|srun|jsrun)(<python[^>]*>(<flux-job>)?|<bash>|<jsrun>|<jswrap>)*$ go}} }
```

If that doesn't work, it looks like you can get a literal block without the code block directive by ending the previous line with `::` and indenting the actual block [[1]](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#literal-blocks):

```suggestion
exec-handling Tcl code into TotalView::

    catch {dset TV::exec_handling { {^(flux|lrun|srun|jsrun)(<python[^>]*>(<flux-job>)?|<bash>|<jsrun>|<jswrap>)*$ go}} }
```

Reply:
I tried with Tcl and it didn't work. I will just go with the second suggestion.